internal/command/test.go

```diff
@@ -235,14 +235,20 @@ func (c *TestCommand) Run(rawArgs []string) int {
 			TestingDirectory:    args.TestDirectory,
 			Opts:                opts,
 			View:                view,
-			JUnit:               junitFile,
 			Stopped:             false,
 			Cancelled:           false,
 			StoppedCtx:          stopCtx,
 			CancelledCtx:        cancelCtx,
 			Filter:              args.Filter,
 			Verbose:             args.Verbose,
 		}
+
+		if junitFile != nil {
```

Comment:
If we were to delay the creation of the junit file until here (instead of creating it above) we could provide the runner in the constructor and avoid the need for the whole `SetTestSuiteRunner` function. Does that work?

Reply:
Yeah that's much better - once the JUnit code was no-longer related to Views it didn't have any reason to remain in that previous position. Addressed in https://github.com/hashicorp/terraform/pull/36315/commits/0f1e8fde33cd56b9cafc116087ef9f284f12cbe5